File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        jumpRange = nums[0]

        for i in range(1, len(nums)):
            if jumpRange <= 0:
                return False
            jumpRange -= 1

            jumpRange = max(jumpRange, nums[i])

        return True

File: Data Structures & Algorithms/sort-colors/submission-1.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int redP = 0, blueP = nums.size()-1;

        for(int i = redP; i <= blueP; i++){
            while(redP <= blueP && nums[redP] == 0) redP++;
            while(blueP > redP && nums[blueP] == 2) blueP--;
            
            if(i < redP) continue;
            if(blueP <= redP) return;

            if(nums[i] == 0){
                nums[i] = nums[redP];
                nums[redP++] = 0;
                i--;
            }
            else if(nums[i] == 2){
                nums[i] = nums[blueP];
                nums[blueP--] = 2;
                i--;
            }
        }
    }
};